tcc: error: undefined symbol 'anon_fn_placeholder_placeholder_172'

**V version:** V 0.2.2 fe88a44
**OS:** windows, Microsoft Windows 10 Pro v19042 64-bit

**What did you do?**
Minimum repro:
```v
import ui

struct App {
mut:
	window &ui.Window = 0
}

fn main() {

	mut app := &App{}
	window := ui.window({
		on_key_down: fn(e ui.KeyEvent, wnd &ui.Window)  // Here would normally be some braces
	},[]
	)

	app.window = window
	ui.run(window)

}
```

**What did you expect to see?**
Some kind of syntax error
 
**What did you see instead?**
```
C:\src\v\thirdparty\stb_image\stbi.o not found, building it in C:\Users\allie\.vmodules\cache\43\43f3c2b198348c8adeb4d539aaaace92.o ...
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:518: warning: WINVER redefined
In file included from C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:918:
In file included from C:/src/v/thirdparty/sokol/sokol_app.h:1354:
c:/src/v/thirdparty/tcc/include/winapi/winapifamily.h:21: warning: WINAPI_FAMILY_PARTITION redefined
In file included from C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:918:
C:/src/v/thirdparty/sokol/sokol_app.h:5941: warning: implicit declaration of function 'freopen_s'
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:7325: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:11009: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:14947: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16235: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16474: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16501: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16581: warning: implicit declaration of function 'GetFinalPathNameByHandleW'
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16665: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16666: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:16683: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:17009: warning: implicit declaration of function 'CreateSymbolicLinkW'
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:17426: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19607: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19617: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19627: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19639: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19671: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19671: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19696: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19698: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19698: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:19723: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:20508: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:20511: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:20515: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:23132: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:23278: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:23281: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:25248: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29052: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29252: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29253: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29254: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29301: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29364: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29365: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29366: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:29891: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.16824363105132326575.tmp.c:30285: warning: assignment from incompatible pointer type
tcc: error: undefined symbol 'anon_fn_placeholder_placeholder_172'
builder error:
==================
C error. This should never happen.
```

Still having issues on latest
```
C:\Users\allie\.vmodules\cache\43\43f3c2b198348c8adeb4d539aaaace92.o is older than C:\src\v\thirdparty\stb_image\stbi.c, rebuilding ...
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:518: warning: WINVER redefined
In file included from C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:915:
In file included from C:/src/v/thirdparty/sokol/sokol_app.h:1354:
c:/src/v/thirdparty/tcc/include/winapi/winapifamily.h:21: warning: WINAPI_FAMILY_PARTITION redefined
In file included from C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:915:
C:/src/v/thirdparty/sokol/sokol_app.h:5941: warning: implicit declaration of function 'freopen_s'
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:7322: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:11006: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:14938: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16226: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16465: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16492: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16572: warning: implicit declaration of function 'GetFinalPathNameByHandleW'
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16656: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16657: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:16674: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:17000: warning: implicit declaration of function 'CreateSymbolicLinkW'
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:17417: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19598: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19608: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19618: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19630: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19662: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19662: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19687: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19689: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19689: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:19714: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:20499: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:20502: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:20506: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:23123: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:23269: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:23272: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:25239: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29043: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29243: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29244: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29245: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29292: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29355: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29356: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29357: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:29882: warning: assignment from incompatible pointer type
C:/Users/allie/AppData/Local/Temp/v/min.14590744271466062812.tmp.c:30276: warning: assignment from incompatible pointer type
tcc: error: undefined symbol 'anon_fn_placeholder_placeholder_211'
builder error:
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.
```

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -324,6 +324,15 @@ pub fn (mut c Checker) struct_init(mut node ast.StructInit) ast.Type {
 							field.pos)
 					}
 				}
+				if field_type_sym.kind == .function && field_type_sym.language == .v {
+					pos := field.expr.pos()
+					if mut field.expr is ast.AnonFn {
+						if field.expr.decl.no_body {
+							c.error('cannot initialize the fn field with anonymous fn that does not have a body',
+								pos)
+						}
+					}
+				}
 				node.fields[i].typ = expr_type
 				node.fields[i].expected_type = field_info.typ
 

--- a/vlib/v/checker/tests/struct_field_init_with_nobody_anon_fn_err.out
+++ b/vlib/v/checker/tests/struct_field_init_with_nobody_anon_fn_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/struct_field_init_with_nobody_anon_fn_err.vv:7:7: error: cannot initialize the fn field with anonymous fn that does not have a body
+    5 | fn main() {
+    6 |     _ = App{
+    7 |         cb: fn(x int)  // Note the missing `{}` (the function body) here
+      |             ~~~~~~~~~
+    8 |     }
+    9 | }

--- a/vlib/v/checker/tests/struct_field_init_with_nobody_anon_fn_err.vv
+++ b/vlib/v/checker/tests/struct_field_init_with_nobody_anon_fn_err.vv
@@ -1,0 +1,9 @@
+struct App {
+	cb fn(x int) // the function signature doesn't make a difference
+}
+
+fn main() {
+	_ = App{
+		cb: fn(x int)  // Note the missing `{}` (the function body) here
+	}
+}

